server/app/mutations/stacks/create.rb

```diff
@@ -12,7 +12,12 @@ class Create < Mutations::Command
     end
 
     optional do
-      string :parent_name
+      hash :parent do
+        required do
+          string :id
```

Comment:
The reason why this wasn't an id in the first place was that children are created before parents, so you don't have an ID (but i suppose names and id's are interchangeable in a lot of models)

Reply:
The `kontena stack install` command is still POST'ing `{"parent_name": ...}`, whereas `kontena stack upgrade` is now PUT'ing `{"parent": {"name": ... } }`.

Not sure what the meaning of the `id` is...

Reply:
This could probably be fixed by using `string :name` instead of `string :id`. That implies that the existence of the `name` stack does not get validated.

Brief reasoning of examples:
* `{"parent": {"name": "foo"}}` is more consistent with the API response format than `{"parent_name": "foo" }`
* `{"parent": {"name": null}}` is certainly wrong
* `{"parent": null}` makes more sense than `{"parent_name": null}`

Reply:
If we're changing this as part of this PR, then the `kontena stack install` commands also needs to be fixed to POST with the new syntax.

Reply:
Using `name` instead of `id` would also be consistent with the response format:

* `{"parent": null}` is a standalone stack without any dependent child stacks
* `{"parent": {"name": "foo", "id": null}}` is an orphaned child stack that does not have its parent
* `{"parent": {"name": "foo", "id": "grid/foo"}}` is a child stack that has an existing parent stack - that is not a valid scenario for `POST /v1/grids/.../stacks`

Reply:
@SpComb yeah, let's change `parent.id` to `parent.name`.